File: sdk/typescript/examples/submit-job.ts
// SPDX-License-Identifier: Apache-2.0

/**
 * Example: Submit a VM export job to HyperSDK
 */

import { HyperSDK, JobDefinition, ExportFormat } from '../src';

async function main() {
  // Initialize client
  const client = new HyperSDK('http://localhost:8080');

  try {
    // Login (if authentication is enabled)
    // await client.login('admin', 'password');

    // Check daemon health
    const health = await client.health();
    console.log(`✓ Daemon is ${health.status}`);

    // Get daemon status
    const status = await client.status();
    console.log(`✓ Version: ${status.version}`);
    console.log(`✓ Uptime: ${status.uptime}`);
    console.log(`✓ Total jobs: ${status.total_jobs}`);
    console.log(`✓ Running jobs: ${status.running_jobs}`);

    // Create job definition
    const jobDef: JobDefinition = {
      name: 'Export Ubuntu Server',
      vm_path: '/Datacenter/vm/ubuntu-server',
      output_dir: '/exports',
      vcenter: {
        server: 'vcenter.example.com',
        username: 'administrator@vsphere.local',
        password: 'your-password',
        insecure: true, // Skip TLS verification for development
      },
      format: ExportFormat.OVF,
      compress: true,
      thin: true,
    };

    // Submit job
    console.log('\nSubmitting job...');
    const jobId = await client.submitJob(jobDef);
    console.log(`✓ Job submitted successfully!`);
    console.log(`  Job ID: ${jobId}`);

    // Get job details
    const job = await client.getJob(jobId);
    console.log(`  Status: ${job.status}`);

    if (job.progress) {
      console.log(`  Progress: ${job.progress.percent_complete}%`);
      console.log(`  Phase: ${job.progress.phase}`);
    }

    console.log(`\nMonitor job progress with:`);
    console.log(`  ts-node examples/monitor-jobs.ts ${jobId}`);
  } catch (error) {
    console.error('✗ Error:', error);
    process.exit(1);
  }
}

main();

File: sdk/python/examples/submit_job.py
#!/usr/bin/env python3
# SPDX-License-Identifier: Apache-2.0

"""Example: Submit a VM export job to HyperSDK."""

from transiva import HyperSDK, JobDefinition, VCenterConfig, ExportFormat

def main():
    # Initialize client
    client = HyperSDK("http://localhost:8080")

    try:
        # Login (if authentication is enabled)
        # client.login("admin", "password")

        # Check daemon health
        health = client.health()
        print(f"✓ Daemon is {health['status']}")

        # Get daemon status
        status = client.status()
        print(f"✓ Version: {status.version}")
        print(f"✓ Uptime: {status.uptime}")
        print(f"✓ Total jobs: {status.total_jobs}")
        print(f"✓ Running jobs: {status.running_jobs}")

        # Define vCenter connection
        vcenter = VCenterConfig(
            server="vcenter.example.com",
            username="administrator@vsphere.local",
            password="your-password",
            insecure=True  # Skip TLS verification for development
        )

        # Create job definition
        job_def = JobDefinition(
            name="Export Ubuntu Server",
            vm_path="/Datacenter/vm/ubuntu-server",
            output_dir="/exports",
            vcenter=vcenter,
            format=ExportFormat.OVF,
            compress=True,
            thin=True
        )

        # Submit job
        print("\nSubmitting job...")
        job_id = client.submit_job(job_def)
        print(f"✓ Job submitted successfully!")
        print(f"  Job ID: {job_id}")

        # Get job details
        job = client.get_job(job_id)
        print(f"  Status: {job.status.value}")

        if job.progress:
            print(f"  Progress: {job.progress.percent_complete}%")
            print(f"  Phase: {job.progress.phase}")

        print(f"\nMonitor job progress with:")
        print(f"  python examples/monitor_jobs.py {job_id}")

    except Exception as e:
        print(f"✗ Error: {e}")
        return 1
    finally:
        client.close()

    return 0


if __name__ == "__main__":
    exit(main())
